### DISCUSSION TITLE
Emails & messages - stuck on loading - Docker

### DISCUSSION THREAD

**Discussion by Couiljer** [saleor/apps#1114] [Q&A] [answered]
Hi there,

First of all, thank you all for the fantastic work!

I've successfully installed the Email & Messages app on a self-hosted instance using Docker and Traefik as a reverse proxy. Everything has been going smoothly so far. I was able to install the app from its URL and access its configuration page from within the dashboard.

However, I've hit a snag. I'm currently stuck with a "loading" message that appears next to the "Messaging Providers" title. No error logs on dashboard console / network, neither from the email&message app container. 

Could anyone provide some guidance or suggestions on how to resolve this issue? Any help would be greatly appreciated!

Thank you in advance for your support.

![Capture d’écran 2023-11-21 à 01 03 59](https://github.com/saleor/apps/assets/16815611/3359ca34-f18a-41a6-8a05-bfc16753eb78)

**Comment by lkostrowski** [comment]
Hey

Please add `APP_LOG_LEVEL=trace` and run the app in development mode. Some logs should be visible in the container

**Comment by Couiljer** [comment, marked as answer]
Thanks for your help @lkostrowski ! 

Everything worked fine in development, and the logs indicated that I **didn't specify the `SECRET_KEY` environment variable** needed for production. **Adding it in production solved my issue !** 

I misunderstood this sentence [ from the documentation](https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-apps/app-examples#saleor-apps) about `SECRET_KEY`: "If not set, a random key will be generated on each app start.". It appears to be true only in development mode.

**Reply by lkostrowski** [reply]
Great that it helped. We will revisit the docs and clarify this
